python/cudf/cudf/core/series.py

```diff
@@ -1680,11 +1693,12 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         codes = Series(cudautils.arange(len(cats), dtype=dtype))
 
         value = DataFrame({"value": cats, "code": codes})
-        codes = DataFrame({"value": self, "order": order})
+        codes = DataFrame({"value": self.copy(), "order": order})
```

Comment:
Why is the copy now needed?

Reply:
This is a shallow copy to protect `self` from any mutations that may happen in the `merge()`

Reply:
Ah ok. Was just curious since it got added in this PR and wasn't there before. Happy to defer 🙂

Reply:
I'm pretty sure what prompted this change is the fact that we now store names of a DataFrame exclusively as the keys of a dictionary behind  the DataFrame.

Before we had column names stored both in the OrderedDict as well as attributes of the individual columns.